cdviz-collector/src/sinks/http.rs

```diff
@@ -44,20 +46,29 @@ impl HttpSink {
 impl Sink for HttpSink {
     //TODO use cloudevents
     async fn send(&self, msg: &Message) -> Result<()> {
-        let json = serde_json::to_value(&msg.cdevent)?;
-        let resp = self
-            .client
-            .post(self.dest.clone())
-            .json(&json)
-            .send()
-            .await?;
-        if !resp.status().is_success() {
-            tracing::warn!(
-                cdevent = ?msg.cdevent,
-                http_status = ?resp.status(),
-                "failed to send event"
-            )
-        }
+        let cd_event = msg.cdevent.clone();
+        // convert  CdEvent to cloudevents
+        let event_result = EventBuilderV10::new().with_cdevent(cd_event).unwrap().build();
+
+        match event_result {
+            Ok(value) => {
+                let resp = self
+                    .client
+                    .post(self.dest.clone())
+                    .json(&value)
```

Comment:
it's not the way to send cloudevent over HTTP.
Part of data is send via HTTP Headers and a part via body.
You could use the provided integration between cloudevents and request: https://github.com/cloudevents/sdk-rust/blob/c4a5443d19a8333104428213e1a4c1a7d2e3adca/example-projects/reqwest-wasm-example/src/lib.rs